### PR TITLE
[JENKINS-69257] Model link chevron is not in center in user build cause

### PR DESCRIPTION
**Pull request by ridemountainpig**
See [JENKINS-69257](https://issues.jenkins.io/browse/JENKINS-69257).

### After
截圖 2022-08-05 下午3 42 20
截圖 2022-08-05 下午3 42 38

### Proposed changelog entries

* Model link chevron is not in center in user build cause

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")` if applicable.
- [ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease future introduction of Content-Security-Policy directives (see [documentation on jenkins.io](https://www.jenkins.io/doc/developer/security/csp/)).
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/6970"></a>

### CODE REVIEW

**Review by NotMyFault — COMMENTED** [review]
Untested, but the change proposed looks like it places existing chevrons lower and therefore dealigns them from being centered

Edit: Yes it does:
Screenshot 2022-08-05 at 11 43 49

**Comment by ridemountainpig** [comment]
> Untested, but the change proposed looks like it places existing chevrons lower and therefore dealigns them from being centered

截圖 2022-08-05 下午9 35 11

截圖 2022-08-05 下午9 37 35

I have using without adding margin-top in jenkins-menu-dropdown-chevron to fix one in this issue, but another one I couldn't find where to fix it, if anyone have any idea welcome give the comment.

File: core/src/main/java/hudson/console/ModelHyperlinkNote.java
Comment (daniel-beck) [inline core/src/main/java/hudson/console/ModelHyperlinkNote.java:30]: This is not a good solution, as this markup is added to the build log as the encoded link is written to it.

This means that only build logs from after a Jenkins update will include these enhancements.

There needs to be a different solution for build logs to adapt to constant style changes than changing the actual log output that is written.

Comment (ridemountainpig) [reply core/src/main/java/hudson/console/ModelHyperlinkNote.java:30]: Sorry, do you have any better idea to resolve this issue, because I only found this way to solve it ?
Thanks~

Comment (daniel-beck) [reply core/src/main/java/hudson/console/ModelHyperlinkNote.java:30]: No idea. If there are CSS classes we can apply to the entire build log and then apply hierarchical rules, that might work.

Otherwise pinging @janfaracik how he imagined this to work.

Comment (basil) [reply core/src/main/java/hudson/console/ModelHyperlinkNote.java:30]: > This means that only build logs from after a Jenkins update will include these enhancements, and old build logs will continue to look broken.

@daniel-beck Does this PR make anything worse? As far as I can tell, it is only an improvement over the status quo, in that new build logs from after a Jenkins upgrade will include this fix. True, a more comprehensive solution could be written, but nobody has volunteered to do that. In fact, the present author even closed this PR! If this change doesn't make anything else worse, but it offers an improvement for new build logs, I think it is worth reopening and merging this PR as-is.

Comment (ridemountainpig) [reply core/src/main/java/hudson/console/ModelHyperlinkNote.java:30]: @basil Sorry, I have reopen this PR. Thanks~

**Review by basil — COMMENTED** [review]
The question I asked in https://github.com/jenkinsci/jenkins/pull/6970#discussion_r959006975 remains unacknowledged.

File: core/src/main/java/hudson/console/ModelHyperlinkNote.java
Comment (daniel-beck) [reply core/src/main/java/hudson/console/ModelHyperlinkNote.java:30]: > This is not a good solution, as this markup is added to the build log as the encoded link is written to it.

This statement is wrong. The markup is not part of the serialized form of the object I apologize for getting this wrong and causing a lot of delay here.

**Comment by daniel-beck** [comment]
> The question I asked in [#6970 (comment)](https://github.com/jenkinsci/jenkins/pull/6970#discussion_r959006975) remains unacknowledged.

My feedback was wrong and the problem I expected does not actually occur.

**Comment by basil** [comment]
Confirmed that this fixes [JENKINS-69257](https://issues.jenkins.io/browse/JENKINS-69257) as follows:

- Checked out `jenkins-2.320`
- Confirmed that `git grep jenkins-table__link` returned no results
- Compiled the code
- Built a freestyle project and confirmed the chevron was at the center line
- Checked out the latest tip-of-trunk with the changes from this PR
- Compiled the code
- Looked at the existing build from before the upgrade, verified that chevron was aligned in the middle

Confirmed that this does not fix [JENKINS-69602](https://issues.jenkins.io/browse/JENKINS-69602), which will remain open after this PR is merged.

**Review by basil — APPROVED** [review]

**Review by NotMyFault — APPROVED** [review]
Thanks for addressing the feedback!

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
